cli/js/40_testing.js

```diff
@@ -677,11 +677,21 @@ function test(
   // Delete this prop in case the user passed it. It's used to detect steps.
   delete testDesc.parent;
   const jsError = core.destructureError(new Error());
-  testDesc.location = {
-    fileName: jsError.frames[1].fileName,
-    lineNumber: jsError.frames[1].lineNumber,
-    columnNumber: jsError.frames[1].columnNumber,
-  };
+  let location;
+
+  for (let i = 0; i < jsError.frames.length; i++) {
+    const filename = jsError.frames[i].fileName;
+    if (filename.startsWith("ext:") || filename.startsWith("node:")) {
+      continue;
+    }
+    location = {
+      fileName: jsError.frames[i].fileName,
+      lineNumber: jsError.frames[i].lineNumber,
+      columnNumber: jsError.frames[i].columnNumber,
+    };
+    break;
+  }
+  testDesc.location = location;
```

Comment:
Function is getting large. In the future, would be good to break this up a bit probably:

```js
testDesc.location = getLocationFromError(jsError)
```

Reply:
I agree